src/components/20-molecules/image-upload/index.js

```diff
@@ -0,0 +1,433 @@
+  }
+
+  static get properties() {
+    return {
+      inputFileText: { type: String },
+      maxSizeOfSingleFileKB: { type: Number },
+      maxSizeOfAllFilesKB: { type: Number },
+      maxNumberOfFiles: { type: Number },
+      showImageOverview: { type: Boolean },
+      icon: { type: String },
+      deleteStatusText: { type: String },
+      addStatusText: { type: String },
+      fileTooBigStatusText: { type: String },
+      filesTooBigStatusText: { type: String },
+      tooManyFilesStatusText: { type: String },
+      orText: { type: String },
+      infoText: { type: String },
+    };
+  }
+
+  constructor() {
+    super();
+    this.inputFileText = 'Upload file';
+    this.maxSizeOfSingleFileKB = 100;
+    this.maxSizeOfAllFilesKB = 500;
+    this.maxNumberOfFiles = 10;
+    this.showImageOverview = false;
+    this.icon = 'cloud-upload';
+    this.deleteStatusText = 'Delete';
+    this.addStatusText = 'Add more';
+    this.fileTooBigStatusText = `File size exceeds maximum size`;
+    this.filesTooBigStatusText = `File sizes exceed maximum size`;
+    this.tooManyFilesStatusText = `You exceeded the maximum number of files`;
+    this.orText = 'or';
+    this.infoText = 'Drag and drop to upload your file';
+    this.files = [];
+    this.faultyFiles = [];
+    this.allFiles = [];
+
+    this.sizeOfAllFilesByte = 0;
+    this.allDroppedFiles = 0;
+    this.isFileMaxReached = false;
+
+    this.globalErrorMessage = '';
+    this.addMoreInputFile = '';
+    this.showAddMoreInputFile = '';
+  }
+
+  handleAddMoreInputClick() {
+    /* trigger inputFile component after clicking the "fake" inputFile (addMoreInput) */
+    this.inputFile.querySelector('input').click();
+  }
+
+  handleImageUploadButtonChange(e) {
+    const { files } = e.target;
+    this.addFiles(files);
+  }
+
+  handleImageUploadDropZoneDragover(e) {
+    /* prevent default browser behavior to execute the link that comes with the event */
+    e.preventDefault();
+    if (!this.isFileMaxReached) {
+      e.dataTransfer.dropEffect = 'copy';
+      this.dropZone.classList.add('m-image-upload__dropzone_dragover');
+    }
+  }
+
+  handleImageUploadDropZoneDragleave() {
+    this.dropZone.classList.remove('m-image-upload__dropzone_dragover');
+  }
+
+  handleImageUploadDropZoneDrop(e) {
+    /* prevent browser to display the file fullscreen */
+    e.preventDefault();
+
+    if (this.isFileMaxReached) {
+      return;
+    }
+    const files = [...e.dataTransfer.files].filter(
+      file => ACCEPTED_FILE_TYPES.indexOf(file.type) > -1
+    );
+
+    this.dropZone.classList.remove('m-image-upload__dropzone_dragover');
+    if (files.length > 0) {
+      this.addFiles(files);
+    }
+  }
+
+  handleImageDeletion(index) {
+    let clonedFiles = [];
+
+    this.allDroppedFiles =
+      (this.files.length === this.maxNumberOfFiles
+        ? this.maxNumberOfFiles
+        : this.allDroppedFiles) - 1;
+
+    this.globalErrorMessage = '';
+
+    if (index >= this.files.length) {
```

Comment:
The reason is that in `this.files` are only valid files. But faulty files are also displayed. Because the faulty files are displayed after the valid ones the index of a faulty files is bigger than the max index aka length of the file array.